var/spack/repos/builtin/packages/root/package.py

```diff
@@ -326,6 +326,79 @@ def determine_version(cls, exe):
         # into the format used in this recipe (6.22.06)
         return output.strip().replace('/', '.')
 
+    @classmethod
+    def determine_variants(cls, exes, version_str):
+        variants = []
+        # there is a fairly direct mapping between build options ( which
+        # which root-config helpfully outputs) and variants
+        output = Executable('root-config')('--features', output=str, error=str)
```

Comment:
You should use exes here, as there may be multiple in the PATH

Reply:
Ah, I was a bit confused by that initially since the interface is different from determine_version, but I think I understand it now.